pkg/api/dock.go

```diff
@@ -59,17 +59,21 @@ func (this *DockPortal) ListDocks() {
 	return
 }
 
-type SpecifiedDockPortal struct {
-	beego.Controller
-}
-
-func (this *SpecifiedDockPortal) GetDock() {
+func (this *DockPortal) GetDock() {
 	id := this.Ctx.Input.Param(":dockId")
 
 	result, err := db.C.GetDock(id)
 	if err != nil {
-		reason := fmt.Sprintf("Get docks failed: %s", err.Error())
-		this.Ctx.Output.SetStatus(StatusBadRequest)
+		reason := fmt.Sprintf("Get dock failed: %s", err.Error())
+		this.Ctx.Output.SetStatus(StatusInternalServerError)
+		this.Ctx.Output.Body(utils.ErrorStatus(this.Ctx.Output.Status, reason))
+		log.Error(reason)
+		return
+	}
+
+	if result == nil {
```

Comment:
Honestly, I don't think it's neccesary to add this sentence, for the reason that every time the result is nil, the err must be ```not nil```, so it can be assured that the result is ```not nil``` by asserting if err is equal to nil.